### PR TITLE
DAT-3994 add checkIfTemplateExists function

### PR DESCRIPTION
**Pull request by dianafa**
mercury part: https://github.com/Wikia/mercury/pull/2243
(this PR has to be released before mercury part)

https://wikia-inc.atlassian.net/browse/DAT-3994

### CODE REVIEW

File: extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php [hunk 1]
```diff
@@ -69,6 +69,17 @@ public function getRedirectUrls() {
 		}
 	}
 
+	public function checkIfTemplateExists() {
+		$status = new Status();
+		$response = $this->getResponse();
+		$response->setFormat( WikiaResponse::FORMAT_JSON );
+		$requestParams = $this->getRequest()->getParams();
+		$title = PortableInfoboxBuilderHelper::getTitle( $requestParams[ 'title' ], $status );
```
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php]: You could use $this->getRequest->getVal('title') without creating the supporting value

File: extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php [hunk 2]
```diff
@@ -69,6 +69,17 @@ public function getRedirectUrls() {
 		}
 	}
 
+	public function checkIfTemplateExists() {
+		$status = new Status();
+		$response = $this->getResponse();
```
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php:75]: I think we can group processing of the request at the beginning and creation of response object along with setting of its vars at the end of this method

File: extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php [hunk 3]
```diff
@@ -69,6 +69,17 @@ public function getRedirectUrls() {
 		}
 	}
 
+	public function checkIfTemplateExists() {
```
Comment (sebastianmarzjan) [inline extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php]: What would be a good alternative to this name, emphasizing that it is a 'get'?
Maybe getTemplateExistenceStatus or getTemplateStatus?

File: extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php [hunk 4]
```diff
@@ -69,6 +69,15 @@ public function getRedirectUrls() {
 		}
 	}
 
+	public function getTemplateExists() {
+		$title = PortableInfoboxBuilderHelper::getTitle( $this->getRequest()->getVal('title'), new Status() );
+
+		$response = $this->getResponse();
+		$response->setFormat( WikiaResponse::FORMAT_JSON );
+		$response->setVal( 'exists', $title->isKnown() );
```
Comment (idradm) [inline extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php]: Are we sure the title object will be created?

File: extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php [hunk 5]
```diff
@@ -69,6 +69,15 @@ public function getRedirectUrls() {
...
+		$response->setVal( 'success', true );
```
Comment (idradm) [inline extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php]: What's the purpose of this variable?

File: extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php [hunk 6]
```diff
@@ -69,6 +69,21 @@ public function getRedirectUrls() {
...
+
+		$response = $this->getResponse();
+		$response->setFormat( WikiaResponse::FORMAT_JSON );
+
+		if  ( $title ) {
+			$response->setVal( 'exists', $title->isKnown() );
+			$response->setVal( 'success', true );
+		} else {
+			$response->setCode( 400 );
+			$response->setVal( 'errors', [ 'Invalid title string has been passed' ] );
```
Comment (idradm) [inline extensions/wikia/PortableInfoboxBuilder/PortableInfoboxBuilderController.class.php]: Shouldn't it be "Invalid title string was passed" or "Invalid title string passed"?

**Comment by sebastianmarzjan** [comment]
LG

**Comment by dianafa** [comment]
this code needs to be merged to and tested together with DAT-3992. Closing this PR.

**Comment by dianafa** [comment]
New PR: https://github.com/Wikia/app/pull/10096

**Pull request closed without merging**